datalad/support/gitrepo.py

```diff
@@ -3627,3 +3639,33 @@ def _fixup_submodule_dotgit_setup(ds, relativepath):
     lgr.debug(
         'Not retro-fitting GitRepo with deprecated symbols, '
         'datalad-deprecated package not found')
+
+
+def _check_for_win_compat(files):
+    """Check file names for illegal characters or reserved names on Windows
+    
+    In the case that a non-Windows-compatible file is detected, warn users
+    about potential interoperability issues.
+    """
+    for file in files.keys():
+        f = Path(file).stem
+        if f.casefold() in [r.casefold() for r in RESERVED_NAMES_WIN]:
+            lgr.warning(
+                "Interoperability warning: The file name %s is a reserved name "
+                "on Windows systems, and can not be shared with Windows "
+                "machines. Consider renaming.", f)
+        chars = re.findall(ILLEGAL_CHARS_WIN, f)
+        if chars:
+            lgr.warning(
+                "Interoperability warning: The following characters are "
+                "considered illegal on Windows operating systems: %s. "
+                "Consider renaming %s to ensure cross-platform compatibility.",
+                chars, file)
+        # check for file name ending in a dot or space
+        if file.endswith('.') or file.endswith(' '):
+            lgr.warning(
+                "Interoperability warning: Windows systems can not "
+                "receive or work with files that end with spaces or dots. "
+                "Consider renaming %s ", file
+            )
+
```

Comment:
That might lead to a flood of warnings  (e.g. in case of a single illegal char used consistently across many files).  I wonder if it would be better to collect all problematic files for each case, and then report a single warning per each case with a number and a list of relevant files.

Reply:
true, I'll try to think of something more clever

Reply:
please take a look at the current implementation.